PepperDashEssentials/Room/Types/EssentialsHuddleVtc1Room.cs

```diff
@@ -672,6 +692,35 @@ public static void AllRoomsOff()
 				(room as EssentialsHuddleSpaceRoom).RunRouteAction("roomOff");
 		}
 
+
+		/// <summary>
+		/// Setup the external sources for the Cisco Touch 10 devices that support IHasExternalSourceSwitch
+		/// </summary>
+		private void SetCodecExternalSources()
+		{
+			
+			string codecTieLine = "";
+			codecTieLine = ConfigReader.ConfigObject.TieLines.SingleOrDefault(x => x.DestinationKey == VideoCodec.Key).DestinationPort;
+			(VideoCodec as IHasExternalSourceSwitching).ClearExternalSources();
+			(VideoCodec as IHasExternalSourceSwitching).RunRouteAction = RunRouteAction;
+			var srcList =  ConfigReader.ConfigObject.SourceLists.SingleOrDefault(x => x.Key == SourceListKey).Value.OrderBy(kv => kv.Value.Order);;
+
+				foreach (var kvp in srcList)
+				{
+					var srcConfig = kvp.Value;
+					
+					if (kvp.Key != "codecOsd" && kvp.Key != "roomOff")
+					{
+
+						(VideoCodec as IHasExternalSourceSwitching).AddExternalSource(codecTieLine, kvp.Key, srcConfig.PreferredName, PepperDash.Essentials.Devices.Common.VideoCodec.Cisco.eExternalSourceType.desktop);
+						(VideoCodec as IHasExternalSourceSwitching).SetExternalSourceState(kvp.Key, PepperDash.Essentials.Devices.Common.VideoCodec.Cisco.eExternalSourceMode.Ready);

```

Comment:
We might want to genericize these enums and move them out of the Cisco class. They'll probably apply to other codecs, if they have the same sort of API.

Reply:
I would think these are particular to Cisco codecs.I wouldn't expect other codecs to use the same arguments.